server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationParser.java

```diff
@@ -1008,7 +1018,75 @@ private void parseKeyStore(XMLExtendedStreamReader reader, ServerConfigurationBu
       keyStoreBuilder.build();
    }
 
-   private void parseTrustStoreRealm(XMLExtendedStreamReader reader, ServerConfigurationBuilder builder, TrustStoreRealmConfigurationBuilder trustStoreBuilder) throws
+   private void parseTrustStore(XMLExtendedStreamReader reader, ServerConfigurationBuilder builder, TrustStoreConfigurationBuilder trustStoreBuilder) throws
+         XMLStreamException {
+      String[] attributes = ParseUtils.requireAttributes(reader, Attribute.PATH);
+      trustStoreBuilder.path(attributes[0]);
+      String relativeTo = (String) reader.getProperty(Server.INFINISPAN_SERVER_CONFIG_PATH);
+      trustStoreBuilder.relativeTo(relativeTo);
+      boolean credentialSet = false;
+      for (int i = 0; i < reader.getAttributeCount(); i++) {
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = reader.getAttributeValue(i);
+         Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+         switch (attribute) {
+            case PATH:
+               // Already seen
+               break;
+            case PROVIDER:
+               trustStoreBuilder.provider(value);
+               break;
+            case RELATIVE_TO:
+               relativeTo = ParseUtils.requireAttributeProperty(reader, i);
+               trustStoreBuilder.relativeTo(relativeTo);
+               break;
+            case PASSWORD:
+               trustStoreBuilder.password(value.toCharArray());
+               credentialSet = true;
+               break;
+            default:
+               throw ParseUtils.unexpectedAttribute(reader, i);
+         }
+      }
+      Element element = nextElement(reader);
+      if (element == Element.CREDENTIAL_REFERENCE) {
+         if (credentialSet) {
+            throw Server.log.cannotOverrideCredential(Element.TRUSTSTORE.toString(), Attribute.PASSWORD.toString());
+         }
+         String credential = parseCredentialReference(reader, builder);
+         trustStoreBuilder.password(credential.toCharArray());
+         credentialSet = true;
+         element = nextElement(reader);
+      }
+      if (!credentialSet) {
+         throw Server.log.missingCredential(Element.TRUSTSTORE.toString(), Attribute.PASSWORD.toString());
+      }
+      if (element != null) {
+         throw ParseUtils.unexpectedElement(reader, element);
+      }
+      trustStoreBuilder.build();
+   }
+
+   private void parseTrustStoreRealm(XMLExtendedStreamReader reader, ServerConfigurationBuilder builder, TrustStoreRealmConfigurationBuilder trustStoreBuilder, ServerIdentitiesConfigurationBuilder serverIdentitiesConfigurationBuilder) throws XMLStreamException {
+      String name = "trust";
```

Comment:
Should this just be the default value for the attribute in `TrustStoreRealmConfiguration`?

Reply:
Probably, but this affects all realms, so let's postpone this clean up for another PR